src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1201,20 +1203,17 @@ export class NetworkAreaDiagramViewer {
                     if (typeof edgeAngle !== 'undefined') {
                         traversingBusEdgesAngles.push(edgeAngle);
                     }
-                    // redraw only if there is an edge going to another voltage level
-                    if (edge.node1 != edge.node2) {
```

Comment:
I'm not sure to understand why you did remove this? it's to avoid drawing loops again, as we only translate them so far. So in fact, you cannot update the bus for loop branches